src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,178 @@
+
+            break;
+        }
+
+        case "todo": {
+
+            if (inputArr.length <= 1) {
+                throw new DukeException("Please enter the descriptions for your to-do!");
+            }
+
+            // Pre-processing
+            int idx = input.indexOf(" ");
+            String taskInput = input.substring(idx + 1);
+            String[] taskInputArr = taskInput.split("/");
+
+            // Create task
+            String taskName = taskInputArr[0];
+            Task newTask = new ToDo(taskName);
+
+            command = new AddCommand(newTask);
+
+            break;
+        }
+
+        case "deadline": {
+
+            if (inputArr.length <= 1) {
+                throw new DukeException("Please enter the description for your deadline!");
+            }
+
+            // Pre-processing
+            int idx = input.indexOf(" ");
+            String taskInput = input.substring(idx + 1);
+            String[] taskInputArr = taskInput.split("/");
+            String taskName = taskInputArr[0];
+            String dateTime = taskInputArr[1];
+            idx = dateTime.indexOf(" ");
+            dateTime = dateTime.substring(idx + 1);
+
+            // Create task
+            Task newTask = new Deadline(taskName, dateTime);
+
+            command = new AddCommand(newTask);
+
+            break;
+        }
+
+        case "event": {
+
+            if (inputArr.length <= 1) {
+                throw new DukeException("Please enter the description for your event!");
+            }
+
+            // Pre-processing
+            int idx = input.indexOf(" ");
+            String taskInput = input.substring(idx + 1);
+            String[] taskInputArr = taskInput.split("/");
+            String taskName = taskInputArr[0];
+            String dateTime = taskInputArr[1];
+            idx = dateTime.indexOf(" ");
+            dateTime = dateTime.substring(idx + 1);
+
+            // Create task
+            Task newTask = new Event(taskName, dateTime);
+
+            command = new AddCommand(newTask);
+
+            break;
+        }
+
+        case "find": {
+
+            if (inputArr.length <= 1) {
+                throw new DukeException("Please enter a keyword so that we can retrieve the appropriate task(s)!");
+            }
+
+            // Pre-processing
+            int idx = input.indexOf(" ");
+            String keyword = input.substring(idx + 1);
+
+            command = new FindCommand(keyword);
+
+            break;
+        }
+
+        case "urgent": {
+
+            // Pre-processing
+            int idx = input.indexOf(" ");
+            String keyword = input.substring(idx + 1);
+
+            command = new UrgentCommand();
+
+            break;
+        }
+
+        case "list": {
+            command = new ListCommand();
+            break;
+        }
+
+        case "bye": {
+            command = new ByeCommand();
+            break;
+        }
+
+        default: {
```

Comment:
You don't need to have a new line between each [`case` statement](https://nus-cs2103-ay1920s2.github.io/website/coding-standards/java/intermediate.html#layout) :)